Assets/Scripts/Manager/GhostSwitchManager.cs

```diff
@@ -116,6 +116,10 @@ public void updateDolls ()
 		}
 	}
 
+	void updateAudio(){
+		MechanicAudioManager.getInstance ().updateAudioControllers ();
```

Comment:
What's the necessity to call updateAudioControllers through the MechanicAudioManager?
You should just use a regular Update() function in the AudioController.

This might be because I said you have to use an audioManager, but you can scratch that cause the AudioController is kind of an audioManager of it's own. This just seems to create more unnecessary. coupling